### PR TITLE
Ensure GCPIngressParams CRD

### PR DESCRIPTION
**Pull request by swetharepakula**
- only ensure if IngressClass resource type exists on cluster
 - ensure GCPIngressParams as a cluster scoped CRD
 - create GCPIngressParams and IngressClass indexers

### CODE REVIEW

File: cmd/glbc/app/init.go [hunk 1]
```diff
@@ -62,6 +63,35 @@ func DefaultBackendServicePort(kubeClient kubernetes.Interface) utils.ServicePor
...
 
+// IngressClassEnabled returns whether the IngressClass API exists on the kubernetes cluster
+func IngressClassEnabled(client kubernetes.Interface) bool {
+	klog.V(2).Info("Checking if Ingress Class API exists")
+
+	err := wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+		var err error
+		_, err = client.NetworkingV1beta1().IngressClasses().List(context.TODO(), meta_v1.ListOptions{})
+		if err == nil {
+			klog.V(4).Info("Ingress Class API is supported")
```
Comment (freehan) [inline cmd/glbc/app/init.go:74]: v(2)

File: cmd/glbc/app/init.go [hunk 2]
```diff
@@ -62,6 +63,35 @@ func DefaultBackendServicePort(kubeClient kubernetes.Interface) utils.ServicePor
...
+			return true, nil
+		}
+
+		if apierrors.IsNotFound(err) {
+			klog.V(4).Infof("Ingress Class API is not supported: %s", err)
```
Comment (freehan) [inline cmd/glbc/app/init.go:79]: V(2)

File: cmd/glbc/app/init.go [hunk 3]
```diff
@@ -62,6 +63,35 @@ func DefaultBackendServicePort(kubeClient kubernetes.Interface) utils.ServicePor
...
+			return true, err
+		}
+		klog.V(4).Infof("Errored checking for Ingress Class API: %s", err)
```
Comment (freehan) [inline cmd/glbc/app/init.go:82]: V(2)

File: cmd/glbc/app/init.go [hunk 4]
```diff
@@ -62,6 +63,35 @@ func DefaultBackendServicePort(kubeClient kubernetes.Interface) utils.ServicePor
...
+		return false, fmt.Errorf("errored checking for ingress class api: %s", err)
```
Comment (freehan) [inline cmd/glbc/app/init.go:83]: wait will return immediately if you return error.

**Review by freehan — COMMENTED** [review]
Make sure the bootstrapping decisions are logged.

File: cmd/glbc/app/init.go (hunk 4 above)
Comment (swetharepakula) [reply cmd/glbc/app/init.go]: Good catch, should be fixed now.

**Comment by swetharepakula** [comment]
Found a way to check for IngressClass resource without querying for the resource itself. Updated the PR with the new logic. This is ready for another round of review.

**Review by freehan — APPROVED** [review]
/lgtm
/approve

**Pull request merged**